contracts/gateway/src/verifier.rs

```diff
@@ -0,0 +1,109 @@
+use crate::contract::Error;
+use axelar_wasm_std::VerificationStatus;
+use connection_router::state::CrossChainId;
+use connection_router::Message;
+use cosmwasm_std::{to_binary, Addr, QuerierWrapper, QueryRequest, WasmMsg, WasmQuery};
+use error_stack::{Result, ResultExt};
+use serde::de::DeserializeOwned;
+use std::collections::HashMap;
+
+pub struct Verifier<'a> {
+    pub querier: QuerierWrapper<'a>,
+    pub address: Addr,
+}
+
+impl Verifier<'_> {
+    fn query<U: DeserializeOwned + 'static>(
+        &self,
+        msg: &aggregate_verifier::msg::QueryMsg,
+    ) -> Result<U, Error> {
+        self.querier
+            .query(&QueryRequest::Wasm(WasmQuery::Smart {
+                contract_addr: self.address.to_string(),
+                msg: to_binary(&msg).expect("msg should always be serializable"),
+            }))
+            .change_context(Error::QueryVerifier)
+    }
+
+    fn execute(&self, msg: &aggregate_verifier::msg::ExecuteMsg) -> WasmMsg {
+        WasmMsg::Execute {
+            contract_addr: self.address.to_string(),
+            msg: to_binary(msg).expect("msg should always be serializable"),
+            funds: vec![],
+        }
+    }
```

Comment:
Wouldn't it be better to define these 2 methods in the `aggregate_verifier` itself?

Reply:
yep, moved